platform/darwin/src/NSCompoundPredicate+MGLAdditions.mm

```diff
@@ -0,0 +1,51 @@
+#import "NSCompoundPredicate+MGLAdditions.h"
+
+#import "NSPredicate+MGLAdditions.h"
+#import "NSExpression+MGLAdditions.h"
+
+class PredicateEvaluator {
+public:
+    
+    std::vector<mbgl::style::Filter>getFilters (NSArray<NSPredicate *> *predicates) {
+        std::vector<mbgl::style::Filter>filters;
+        for (NSPredicate *predicate in predicates) {
+            filters.push_back(predicate.mgl_filter);
+        }
+        return filters;
+    }
+    
+    mbgl::style::Filter operator()(NSCompoundPredicate *predicate) {
+        switch (predicate.compoundPredicateType) {
+            case NSNotPredicateType: {
+                auto filter = mbgl::style::NoneFilter();
+                filter.filters = getFilters(predicate.subpredicates);
+                return filter;
+            }
+            case NSAndPredicateType: {
+                auto filter = mbgl::style::AllFilter();
+                filter.filters = getFilters(predicate.subpredicates);
```

Comment:
[Yes, it does](https://github.com/mapbox/mapbox-gl-native/blob/6a39cf5aaece81c7a531b12321dd503004cc45b8/include/mbgl/style/filter_evaluator.hpp#L102).